src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/readme.md

```diff
@@ -4,95 +4,364 @@ Reference documentation for the current version of the REST API supported by the
-
-The following shows a sample JSON payload for the request:
+The payload for the request is a JSON document:
 
 ``` json
 {
     "token": "...",
-    "applicationId": "contosoapp"
+    "applicationId": "contosoapp",
+    "applicationVersion": "2018.4",
+    "duration": "PT5M",
+    "metering": [{
+        "type": "cpu",
+        "count": 16
+    }]
+}
+```
+
+| Element            | Required  | Type           | Description                                                                                                                                                                                                                                                                                                                                           |
+|--------------------|-----------|----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| token              | Mandatory | string         | The software entitlement token supplied to the application via environment variable from Azure Batch.                                                                                                                                                                                                                                                 |
+| applicationId      | Mandatory | string         | A unique identifier for the application requesting an entitlement to run. <br/> Samples: contosoapp, application <br/> Application identifiers are lowercase alphanumeric (though comparisons will be case-insensitive), with no punctuation, whitespace or non-alphanumeric characters. <br/> Must be an application entitled by the token provided. |
+| applicationVersion | Optional  | string         | The version of the application making the request. <br/> Sample: 2018.4<br/> Maximum length: 64                                                                                                                                                                                                                                                       |
+| duration           | Mandatory | string         | The duration requested for the initial entitlement. <br/> Specified as an ISO8601 string.<br/> Minimum 5 minutes (PT5M), maximum TBC.                                                                                                                                                                                                                 |
```

Comment:
That's a great reminder we need to discuss what the maximum should be. I had to choose a max value when working on the internal API changes, and went for one hour, but we should definitely talk that over.

I'd hope that we discourage vendors from using much larger durations than the recommended 5 minutes (mainly because this is supposed to be more accurate than billing by VM uptime, and partly because I think it'll make the billing logic simpler), but there may be reasons I'm unaware of why more than one hour might be necessary.

Reply:
I'll change this to reflect the current max from the implementation.